File: internal/ProviderPACTemplates.go
package internal

import (
	"os"
	"path/filepath"

	"github.com/gofiber/fiber/v2/log"
	"github.com/timeforaninja/pacserver/pkg/utils"
)

type pacTemplate struct {
	Filename string `json:"Filename"`
	content  string
}

func readTemplateFiles(relPacDir string) ([]*pacTemplate, error) {
	absPACPath, err := filepath.Abs(relPacDir)
	if err != nil {
		log.Errorf("Invalid Filepath for PACs found: \"%s\": %s", absPACPath, err.Error())
		return nil, err
	}
	files, err := utils.ListFiles(absPACPath)
	if err != nil {
		log.Errorf("Failed to List PAC Files in \"%s\": %s", absPACPath, err.Error())
		return nil, err
	}

	var templates []*pacTemplate

	for _, file := range files {
		fullPath := filepath.Join(absPACPath, file)
		fileBytes, err := os.ReadFile(fullPath)
		if err != nil {
			log.Warnf("Unable to read PAC at \"%s\": %s", fullPath, err.Error())
			continue
		}

		templates = append(templates, &pacTemplate{
			Filename: file,
			content:  string(fileBytes),
		})
	}

	return templates, nil
}

File: internal/ProviderPACTemplates.py
import os
from dataclasses import dataclass
import logging
from typing import List

from pkg.utils.listFiles import list_files

@dataclass
class PACTemplate:
    filename: str
    content: str

def read_template_files(rel_pac_dir: str) -> List[PACTemplate]:
    try:
        abs_pac_path = os.path.abspath(rel_pac_dir)
    except Exception as e:
        logging.error(f'Invalid Filepath for PACs found: "{rel_pac_dir}": {str(e)}')
        raise

    try:
        files = list_files(abs_pac_path)
    except Exception as e:
        logging.error(f'Failed to List PAC Files in "{abs_pac_path}": {str(e)}')
        raise

    templates: List[PACTemplate] = []

    for file in files:
        full_path = os.path.join(abs_pac_path, file)
        try:
            with open(full_path, 'r', encoding='utf-8') as f:
                content = f.read()

            templates.append(PACTemplate(
                filename=file,
                content=content
            ))
        except Exception as e:
            logging.warning(f'Unable to read PAC at "{full_path}": {str(e)}')
            continue

    return templates
